aten/src/ATen/core/DistributionsHelper.h

```diff
@@ -126,6 +126,65 @@ struct uniform_real_distribution {
     T to_;
 };
 
+#define DISTRIBUTION_HELPER_GENERATE_HAS_MEMBER(member)              \
```

Comment:
This is a really big macro. Is there really no way to do this without macro?

Reply:
hm... this macro is 8 loc

Reply:
Apparently, this macro is a version of an old-style detection idiom in C++. In C++17 you could have just use `std::is_detected` to do this, no macros necessary. The way we generally prefer to backport features like this is to backport an implementation of the macro in question (e.g., the reference implementation from cppreference) so that we can easily switch to using the standard library version when we eventually upgrade to C++.

Reply:
If you' re saying that I have to implement `std::is_detected` then can I do it in follow up PR and leave this as is?